src/Calculator/Views/CalculatorProgrammerBitFlipPanel.xaml.cpp

```diff
@@ -185,35 +175,49 @@ void CalculatorProgrammerBitFlipPanel::UpdateCheckedStates()
         return;
     }
 
-    static const wchar_t ch0 = LocalizationSettings::GetInstance().GetDigitSymbolFromEnUsDigit(L'0');
-
-    // Filter any unwanted characters from the displayed string.
-    static constexpr array<wchar_t, 4> unwantedChars = { L' ', Utils::LRE, Utils::PDF, Utils::LRO };
-
-    wstringstream stream;
-    wstring displayValue = Model->BinaryDisplayValue->Data();
-    for (const wchar_t& c : displayValue)
+    m_updatingCheckedStates = true;
+    unsigned int bitIndex = 0;
+    auto it = m_flipButtons.begin();
+    for (bool val : Model->BinaryDigits)
     {
-        if (find(begin(unwantedChars), end(unwantedChars), c) == unwantedChars.end())
+        auto checkbox = *it;
+        if (forceUpdate || checkbox->IsChecked->Value != val)
         {
-            stream << c;
+            checkbox->IsChecked = val;
+            // Only generate the string when the value changed
+            checkbox->SetValue(AutomationProperties::NameProperty, GenerateAutomationPropertiesName(bitIndex, val));
```

Comment:
When testing out your fix for localization, I found I wasn't hitting this codepath when buttons are getting updated.  As a result, I'm not seeing automation name change when a bit gets flipped.    It looks like checkbox->IsChecked->Value has already been updated to val at this point, so the check is always passing.

Reply:
You are right. It's now fixed!